### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by JHeisler**
Additions
- Added/Fixed more Karma tests
- Added RDS and EFS Tests, commented out MockBackend Tests

Issues
- RDS describeTagsById should describe by id error and EFS describeTagsById should describe by id error  are getting an error message saying callback is not a function. I fixed this in EFS Describe by removing the {} in line 13 of the controller. I believe something with passing parameters is causing an issue with the callback. 
- There is still an issue with the Analytics and Login Components, Login is more a logical problem on my part and Analytics seems to be caused by the dialog, providing MAT_DIALOG_DATA with some fake data doesn't work despite it working on the Janitor Dialog (line 20 of janitordialog.component.spec.ts).

### CODE REVIEW

File: src/api/test/rds.js [hunk 1]
```diff
@@ -1,63 +1,227 @@
...
+      const req = http_mocks.createRequest({
+        method: 'GET',
+        url: '/api/rds/describe/tags/db-test',
+        params: {
+          id: "db-test"
+        },
+      });
+
+      res.on('end', () => {
+        res.json(res._getData())
```
Comment (rdcaldwell) [inline src/api/test/rds.js]: remove this line

File: src/api/test/rds.js [hunk 2]
```diff
@@ -1,63 +1,227 @@
...
+        done();
+      });
+
+      RDS_CONTROLLER.describeTagsById(req, res);
+      AWS.restore('RDS');
+    });
+
+    // Todo, callback issue same as efs
+    xit('should get describe by id error', (done) => {
+      AWS.mock('RDS', 'listTagsForResource', (callback) => {
```
Comment (rdcaldwell) [inline src/api/test/rds.js]: AWS.mock('RDS', 'listTagsForResource', ({}, callback) => {

File: src/api/test/efs.js [hunk 3]
```diff
@@ -1,63 +1,225 @@
...
+      const req = http_mocks.createRequest({
+        method: 'GET',
+        url: '/api/efs/describe/tags/id',
+        params: {
+          id: '12'
+        }
+      });
+
+      res.on('end', () => {
+        res.json(res._getData())
```
Comment (rdcaldwell) [inline src/api/test/efs.js]: remove

File: src/api/test/efs.js [hunk 4]
```diff
@@ -1,63 +1,225 @@
...
+        done();
+      });
+
+      EFS_CONTROLLER.describeTagsById(req, res);
+      AWS.restore('EFS');
+    });
+
+    // Todo, something to do with params in controller is causing this?
+    xit('should get describe by id error', (done) => {
+      AWS.mock('EFS', 'describeTags', (callback) => {
```
Comment (rdcaldwell) [inline src/api/test/efs.js]: AWS.mock('EFS', 'describeTags', ({}, callback) => {

**Review by rdcaldwell — CHANGES_REQUESTED** [review]

**Comment by JHeisler** [comment]
Changes done, fixed the callback issues. Thanks

**Review by rdcaldwell — APPROVED** [review]

**Pull request merged**